apps/web/src/components/Layout/AdminLayout/AdminLayout.stories.tsx

```diff
@@ -0,0 +1,82 @@
+import React from "react";
+import { Meta, StoryFn } from "@storybook/react";
+import { faker } from "@faker-js/faker";
+
+import {
+  AuthorizationContext,
+  AuthenticationContext,
+  useAuthentication,
+  useAuthorization,
+  ROLE_NAME,
+  RoleName,
+} from "@gc-digital-talent/auth";
+
+import AdminLayout from "./AdminLayout";
+
+const availableRoles = Object.values(ROLE_NAME);
+
+type AdminLayoutArgs = {
+  isLoggedIn: boolean;
+  roles: RoleName[];
+};
+
+export default {
+  component: AdminLayout,
+  title: "Components/Layout/Admin Layout",
+  args: {
+    isLoggedIn: true,
+    roles: availableRoles,
+  },
+  argTypes: {
+    roles: {
+      control: "check",
+      options: availableRoles,
+    },
+  },
+} as Meta<AdminLayoutArgs>;
+
+const Template: StoryFn<AdminLayoutArgs> = (args) => {
+  const { isLoggedIn, roles } = args;
+  const authenticationState = useAuthentication();
+  const authorizationState = useAuthorization();
+
+  const mockAuthenticationState = React.useMemo(
+    () => ({
+      ...authenticationState,
+      isLoggedIn,
+    }),
+    [isLoggedIn, authenticationState],
+  );
+  const mockAuthorizationState = React.useMemo(
+    () => ({
+      ...authorizationState,
+      roleAssignments: roles.map((roleName) => ({
+        id: faker.string.uuid(),
+        role: {
+          id: faker.string.uuid(),
+          name: roleName,
+        },
+      })),
+    }),
+    [roles, authorizationState],
+  );
+
+  return (
+    <AuthenticationContext.Provider value={mockAuthenticationState}>
+      <AuthorizationContext.Provider value={mockAuthorizationState}>
+        <AdminLayout />
+      </AuthorizationContext.Provider>
+    </AuthenticationContext.Provider>
+  );
+};
+
+export const LoggedIn = Template.bind({});
+LoggedIn.args = {
+  isLoggedIn: true,
```

Comment:
Does this need to be an `arg`? Toggling it in storybook doesn't seem to change anything.

Reply:
Let me see. This may actually be a bug in our logic 😅. This should update the link at the bottom from "Sign in" to "Sign out".

Reply:
Yup! We were using the wrong variable name in the original story and I just blindly copy and pasted it 😬 

Fixed in [fix story authentication state](https://github.com/GCTC-NTGC/gc-digital-talent/pull/8667/commits/76eb58778af834e60f984b856837837a1e678656)